### PR TITLE
Make benchmark more like Stockfish bench 

### PR DESCRIPTION
**Pull request by cn4750**
Add some of the SF bench positions, iterate over them while collecting nodes and times, report time and node count summary at the end.

Final output looks like:
```===========================
Total time (ms) : 430606
Nodes searched  : 27275237
Nodes/second    : 63341
```

### CODE REVIEW

**Comment by jjoshua2** [comment]
You say you added some of the positions? How did you pick which ones? And why does comment say // Same positions as Stockfish uses.

**Comment by cn4750** [comment]
@jjoshua2, all the positions but the final three were added as they are terminal or chess 960 positions.

**Comment by borg323** [comment]
We use the current benchmark mode for other stuff, so removing the existing functionality is not likely to be accepted. I would add a new flag for this, say `--auto` (or anything else you like), so that running `lc0 benchmark --auto` will do it the stockfish way (and we can discuss whether it should default to true later). An alternative is to add a new top level mode, so that you run it with `lc0 benchmark-auto`, but this is probably overcomplicating it.

**Comment by cn4750** [comment]
@borg323, I think that having the benchmark default to a more accurate (and longer) bench makes the most sense from a user perspective. I have enabled old bench behavior by using a flag to search only the starting position. I have changed the dx and opencl check scripts to use this flag. Regarding pgo profile generation, we probably want the larger bench to provide a better profile (unless things start taking too long).

**Comment by mooskagh** [comment]
If implemented, fixes #215

**Comment by cn4750** [comment]
@borg323, how about if I remove the `--starting-position` flag and replace it with a `--num-positions` flag which allows you to test a fixed number of positions from the suite? Setting it to one would test only the starting position and if you wanted a quicker than full benchmark run you could choose whatever is appropriate.

**Comment by borg323** [comment]
That would be perfect - in which case we can also update the checks to take about a minute, with 5-6 different positions.

File: src/benchmark/benchmark.cc [hunk 1]
```diff
@@ -25,6 +25,8 @@
   Program grant you additional permission to convey the resulting work.
 */
 
+#include <numeric>
```
Comment (borg323) [inline src/benchmark/benchmark.cc]: I understand the style used requires this after `#include "benchmark/benchmark.h"` but before `#include "mcts/search.h"`

Comment (cn4750) [reply src/benchmark/benchmark.cc]: Should be addressed by commit 2257352 now.

**Pull request merged**